### PR TITLE
AP_TemperatureSensor: Extend analog sensor backend to 5th order polynomial

### PR DESCRIPTION
**Pull request by rishabsingh3003**
After collecting some data from a analog temperature sensor, a 5th order polynomial tracked the temperature better. 

image

image

### CODE REVIEW

File: libraries/AP_TemperatureSensor/AP_TemperatureSensor_Analog.cpp [hunk 1]
```diff
@@ -56,6 +56,11 @@ const AP_Param::GroupInfo AP_TemperatureSensor_Analog::var_info[] = {
     // @Description: a4 in polynomial of form temperature in deg = a0 + a1*voltage + a2*voltage^2 + a3*voltage^3 + a4*voltage^4
     AP_GROUPINFO("A4", 6, AP_TemperatureSensor_Analog, _a[4], 0),
 
+    // @Param: A5
+    // @DisplayName: Temperature sensor analog 5th polynomial coefficient
+    // @Description: a5 in polynomial of form temperature in deg = a0 + a1*voltage + a2*voltage^2 + a3*voltage^3 + a4*voltage^4 + a5*voltage^5
```
Comment (IamPete1) [inline libraries/AP_TemperatureSensor/AP_TemperatureSensor_Analog.cpp:61]: Need to also update the other param descriptions to include a5.

**Review by IamPete1 — APPROVED** [review]
LGTM. This looks correct. I guess the real question if it's worth the flash cost. From the plots does look like a nice improvement.

**Comment by peterbarker** [comment]
Flash cost is kind of minimal:
```
Board                    AP_Periph  blimp  bootloader  copter  heli  iofirmware  plane  rover  sub
CubeOrange-periph-heavy  *                                                                     
Durandal                            *      *           *       *                 *      *      24
Hitec-Airspeed           *                                                                     
KakuteH7-bdshot                     *      *           *       *                 *      *      16
MatekF405                           *      *           *       *                 *      *      *
Pixhawk1-1M-bdshot                  *                  *       *                 *      *      *
f103-QiotekPeriph        *                                                                     
f303-Universal           *                                                                     
iomcu                                                                *                         
revo-mini                           *      *           *       *                 *      *      *
```

**Comment by ChamathkePerera** [comment]
Hi guys, I would like to comment that this curve is for an NTC thermistor. 
Please see the link below from Texas Instruments that explains the calculation process for NTC and PTC:
[https://www.ti.com/lit/an/sboa537/sboa537.pdf?ts=1721931077955&ref_url=https%253A%252F%252Fwww.google.com%252F](url)
A 4th order would likely be the best solution for PTC but could benefit from a 5th order. NTC definitely require 5th order.

**Pull request merged**